src/huggingface_hub/inference/_providers/_common.py

```diff
@@ -36,8 +37,30 @@
 }
 
 
-def filter_none(d: Dict[str, Any]) -> Dict[str, Any]:
-    return {k: v for k, v in d.items() if v is not None}
+def filter_none(obj: Union[Dict[str, Any], List[Any]]) -> Dict[str, Any]:
```

Comment:
I think we can be more precise here to tell _"if Dict as input, Dict as output, if List as input, List as output"_

```suggestion
@overload
def filter_none(obj: Dict[str, Any]) -> Dict[str, Any]: ...
@overload
def filter_none(obj: List[Any]) -> List[Any]: ...

def filter_none(obj: Union[Dict[str, Any], List[Any]]]) -> Union[Dict[str, Any], List[Any]]:
```

(+ import `typing.overload`)

Reply:
addressed in [aba4227](https://github.com/huggingface/huggingface_hub/pull/3178/commits/aba422721a5795fe1a9917f50411a9794d9ab699)